### PR TITLE
fixes unremovable mutations, fixes weird tail organ shennanigans

### LINKED ISSUE #49758: HARS mutation, despite description, is incurable.
## Round ID: 131607

I jabbed myself with a HARS injector, having read the description that it was curable (but might kill me in the process). I took mutadone, nothing happened. I had it removed in the mutation scanner. Nothing happened (and it came back spontaneously later). Even with the mutation 'gone' from the mutation scanner, i was still unable to get a new head via robotics, either.

The HARS was not, as far as I could tell, reinforced.

Being injected with HARS essentially permanently and incurably both blinds and deafens you and makes you unable to be rebodied or borged or possibly even revived due to no longer having a brain.

Even an aheal only restored sight and sound, but i still had no head and as such could not do things like wear a radio.

### LINKED ISSUE #49762: Genetic Mutations persisting through deactivation/nullification/DNA swaps.
## Round ID:
131613

## Testmerges:
None?

## Reproduction:
Genetics. Use the console to amp up some monkey-human action. Break the monkey mutation, everything works as expected. Then later on in the round, start noticing weird behaviour. Mutation visual effects persisting through DNA Scrambles and through breaking Mutation chromosones. Former monkey test subject retaining Radioactive glow despite mutation not being present, does not appear to be gaining radiation percent despite the glow being active.

Conduct self-tests. Load myself up with a ton of mutations and Scramble my DNA. Certain mutations with permanent effects persisted including: Telepathy, X-Ray Vision, Antenna, Transcendent Olfaction. I was able to actively use Telepathy and X-Ray Vision for the entire round despite no longer having them as mutations thanks to scrambling my own DNA.

Most concerning was that I activated genetic Blindness on myself and this ALSO persisted through DNA Scrambles and had to be removed by admin.

However, while some **visual** effects were permanent, such as Chameleon's semi-transparency, when I scrambled my DNA, I would lose the Chameleon invisilibility power as expected.

I did not get a chance to test Space Adaptation although again the visual effect persisted between DNA Scrambles when I no longer had the mutation.

I opened an ahelp with JusticeGoat around the time I noticed this was happenning.

Ultimately, to produce the bug I would give myself or a test candidate a mutation, then nullify said mutation or scramble the DNA. Certain visual effects would still remain and some mutations now became permanent and were unable to be removed, including heavily debilitating effects such as Blindness.

Almost certainly related to Issue #49758

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

fixes #49762
fixes #49758

The mutant organs would override the last mutant organs, dropping them to the floor

### CODE REVIEW

**Comment by tralezab** [comment]
image

**Comment by Fikou** [comment]
no

**Comment by PKPenguin321** [comment]
How does this compare to https://github.com/tgstation/tgstation/pull/49771 ?

**Comment by tralezab** [comment]
that's this pr, but i assume there's another fix already up. let me take a look and i'll be able to tell.

**Comment by tralezab** [comment]
okay, let me give a self review on why this is the better fix.

File: code/datums/mutations/_mutations.dm [hunk 1]
```diff
@@ -93,7 +93,7 @@
 	return
 
 /datum/mutation/human/proc/on_losing(mob/living/carbon/human/owner)
-	if(owner && istype(owner) && (owner.dna.mutations.Remove(src)))
+	if(!owner || !istype(owner) || !(owner.dna.mutations.Remove(src)))
```
Comment (tralezab) [inline code/datums/mutations/_mutations.dm]: what broke mutations is that i added an early return for readability WITHOUT reversing this if check to TRUE on fail. in the other pr, they return TRUE on a completed on_losing when TRUE is supposed to be for failed removals.

File: code/datums/mutations/body.dm [hunk 2]
```diff
@@ -473,11 +473,11 @@
 	. = ..()
 	if(.)
 		return TRUE
+	UnregisterSignal(owner, COMSIG_LIVING_ATTACH_LIMB)
 	var/successful = owner.regenerate_limb(BODY_ZONE_HEAD, noheal = TRUE) //noheal needs to be TRUE to prevent weird adding and removing mutation healing
 	if(!successful)
 		stack_trace("HARS mutation head regeneration failed! (usually caused by headless syndrome having a head)")
 		return TRUE
-	UnregisterSignal(owner, COMSIG_LIVING_ATTACH_LIMB)
```
Comment (tralezab) [inline code/datums/mutations/body.dm]: extra fix in this pr not handled by the other

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -201,9 +201,11 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
 				QDEL_NULL(I)
 
-	for(var/path in mutant_organs)
-		var/obj/item/organ/I = new path()
-		I.Insert(C)
+	for(var/thing in mutant_organs)
+		var/current_organ = C.getorgan(thing)
+		if(!current_organ || replace_current)
+			var/obj/item/organ/missed = new thing()
+			missed.Insert(C, TRUE, FALSE)
```
Comment (tralezab) [inline code/modules/mob/living/carbon/human/species.dm]: big extra fix here not included in the other pr, this is solving mutant species dropping their mutant organs on spawn.

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 4]
```diff
@@ -80,20 +80,18 @@
...
 	if(tail)
 		var/obj/item/organ/tail/NT
-		NT = locate(/obj/item/organ/tail) in new_species.mutant_organs
-		if(NT)
-			// Roundstart cat tail overrides new_species tail, reset it here.
-			new_species.mutanttail = initial(new_species.mutanttail)
-			if(new_species.mutanttail)
-				NT = new new_species.mutanttail
+		// Roundstart cat tail overrides new_species tail, reset it here.
+		if(new_species)
```
Comment (tralezab) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm]: final big fix n shit for roundstart cats

**Review by tralezab — COMMENTED** [review]
why this is preferred to #49767

**Comment by ATH1909** [comment]
(unironically) speedmerge

File: code/datums/mutations/_mutations.dm (hunk 1 above)
Comment (AnturK) [inline code/datums/mutations/_mutations.dm]: no need for !owner, istype does it already

File: code/modules/mob/living/carbon/human/species.dm [hunk 5]
```diff
@@ -189,7 +189,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 
 		if(!oldorgan && should_have && !(initial(neworgan.zone) in excluded_zones))
 			used_neworgan = TRUE
-			neworgan.Insert(C)
+			neworgan.Insert(C, TRUE, FALSE)
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species.dm]: Why these arguments ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/datums/mutations/_mutations.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/mutations/_mutations.dm]: will do

**Pull request merged**